source/agora/common/EnrollmentManager.d

```diff
@@ -205,6 +205,66 @@ public class EnrollmentManager
         this.db.execute("DELETE FROM validator_set WHERE key = ?", enroll_hash[]);
     }
 
+    /***************************************************************************
+
+        In validatorSet DB, return the enrolled block height.
+
+        Params:
+            enroll_hash = key for an enrollment blockHeight
+
+        Returns:
+            the enrolled block height.
+
+    ***************************************************************************/
+
+    public size_t getEnrollmentBlockHeight (const ref Hash enroll_hash) @safe
+    {
+        return () @trusted {
+            return this.db.execute(
+                "SELECT block_height FROM validator_set WHERE key = ?",
+                enroll_hash[]).oneValue!size_t;
+        }();
+    }
+
+    /***************************************************************************
+
+        Update the blockHeight of the validatorSet DB.
+
+        Params:
+            enroll_hash = enrollment blockHeight to update enroll hash
+            block_Height = enrolled blockheight
+
+        Returns:
+            Update execution status
+
+    ***************************************************************************/
+
+    public bool updateEnrollmentBlockHeight (const ref Hash enroll_hash,
+        const size_t block_Height) @safe
+    {
+        size_t enrolled_hight = getEnrollmentBlockHeight(enroll_hash);
+        try
+        {
+            if (enrolled_hight > 0)
+            {
+                return false;
+            }
+            else
+            {
+                () @trusted {
+                    this.db.execute(
+                        "UPDATE validator_set SET block_height = ? WHERE key = ?",
+                        block_Height, enroll_hash[]);
+                }();
+            }
+        }
+        catch (Exception ex)
+        {
+            return false;
```

Comment:
A logging is needed. You could the function called `logMessage` of the EnrollmentManager class.